digital_asset_types/src/dao/mod.rs

```diff
@@ -197,6 +125,36 @@ impl SearchAssetsQuery {
             )
             .add_option(self.burnt.map(|x| asset::Column::Burnt.eq(x)));
 
+        if let Some(s) = self.supply {
+            conditions = conditions.add(asset::Column::Supply.eq(s));
+        } else {
+            // By default, we ignore malformed tokens by ignoring tokens with supply=0
+            // unless they are burnt.
+            //
+            // cNFTs keep supply=1 after they are burnt.
+            // Regular NFTs go to supply=0 after they are burnt.
+            conditions = conditions.add(
+                asset::Column::Supply
+                    .ne(0)
+                    .or(asset::Column::Burnt.eq(true)),
+            )
+        }
+
+        if let Some(o) = self.owner_type.clone() {
+            conditions = conditions.add(asset::Column::OwnerType.eq(o));
+        } else {
+            // Default to NFTs
+            //
+            // In theory, the owner_type=single check should be sufficient,
+            // however there is an old bug that has marked some non-NFTs as "single" with supply > 1.
+            // The supply check guarentees we do not include those.
+            conditions = conditions.add_option(Some(
+                asset::Column::OwnerType
+                    .eq(OwnerType::Single)
+                    .and(asset::Column::Supply.lte(1)),
```

Comment:
If trees and accounts are backfilled will they be corrected?

Related patch for mis-labled owner type for non-NFTs
 https://github.com/metaplex-foundation/digital-asset-rpc-infrastructure/pull/151

Reply:
Yes, a fresh deploy of DAS should not face this issue but I kept it for safety (and since we run it on our system).